mediapipe/tasks/ios/text/core/sources/MPPBaseTextTaskApi.h

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import <Foundation/Foundation.h>
+
+#include "mediapipe/framework/calculator.pb.h"
+#include "mediapipe/tasks/cc/core/task_runner.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * The base class of the user-facing iOS mediapipe text task api classes.
+ */
+NS_SWIFT_NAME(BaseTextTaskApi)
+@interface MPPBaseTextTaskApi : NSObject {
+ @protected
+  std::unique_ptr<mediapipe::tasks::core::TaskRunner> cppTaskRunner;
```

Comment:
@schmidt-sebastian @khanhlvg @lu-wang-g
I am using the C++ task runner like the Python implementation. Java on the other hand, creates its own task runner with the graph APIs. 
If need be, we can follow the Java approach. Only thing is, we will have to rewrite  [MPPGraph](https://github.com/google/mediapipe/blob/master/mediapipe/objc/MPPGraph.h#L25) and add a lot of other classes to process packets etc. The current iOS graph APIs depend on `MPPGraph`. If we use this approach, we will have to ensure, we don't break anything.

LMK if sticking to the Python approach is fine.

Reply:
I think we will be much better off in the long run if we follow the Python approach and re-use as much code as possible.